### ISSUE TITLE
curl: (56) Recv failure: Connection reset by peer

### ISSUE THREAD

**Issue by saparaj** [fabric8io/kubernetes-client#2439]
Hi,

I am trying to create deployment and service using below code - This works first time, if I do curl on service external endpoint I get the reply back. If I update the yaml with updated docker image say: v2 and try curl again, I am getting this error: curl: (56) Recv failure: Connection reset by peer. If I check k8 service logs before and after 2nd deployment, NodePort and Service Endpoints are different. If I try this same scenario using kubectl CLI it works fine. Please let me know what I am doing wrong from code.

`List<HasMetadata> allResources = client.load(stream).get();
client.resourceList(allResources).inNamespace("default").createOrReplaceAnd().waitUntilReady(5, TimeUnit.MINUTES);`

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: helloworld-deployment
spec:
  selector:
    matchLabels:
      app: helloworld
  replicas: 1
  template:
    metadata:
      labels:
        app: helloworld
    spec:
      containers:
        - name: helloworld
          # enter the path to your image, be sure to include the correct region prefix
          image: iad.ocir.io/ax022wvgmjpq/helloworld:v1
          ports:
            - containerPort: 8888
---
apiVersion: v1
kind: Service
metadata:
  name: helloworld-service
spec:
  type: LoadBalancer
  ports:
    - port: 8080
      protocol: TCP
      targetPort: 8888
  selector:
    app: helloworld
```

Thanks,
Swarda

**Comment by rohanKanojia** [comment]
After a discussion on Gitter, I think I understand the issue now. Problem is that client is doing PATCH with YAML object even after there are no new changes in `Service`, but while doing createOrReplace we compare object  provided with object fetched from the server(which is always different since `nodePort` and `clusterIP` are dynamically allocated upon first creation). See below code:
```
try (KubernetesClient client = new DefaultKubernetesClient()) {
    Service svc = new ServiceBuilder()
            .withNewMetadata().withName("my-service").endMetadata()
            .withNewSpec()
            .withType("NodePort")
            .addToSelector("app", "MyApp")
            .addNewPort()
            .withPort(80)
            .withTargetPort(new IntOrString(80))
            .endPort()
            .endSpec()
            .build();

    Service svc1 = client.services().inNamespace("rokumar").createOrReplace(svc);
    // Output: 30614
    System.out.println(svc1.getSpec().getPorts().get(0).getNodePort());
    Service svc2 = client.services().inNamespace("rokumar").createOrReplace(svc);
    // Output: 32290
    System.out.println(svc2.getSpec().getPorts().get(0).getNodePort());
}
```

**Comment by saparaj** [comment]
Hi @rohanKanojia,

Is there any timeline for this issue fix?

Thanks,
Swarda

**Comment by rohanKanojia** [comment]
I would try to fix it along with #2399

**Comment by manusa** [comment]
Discussion for this issue is open in #2454. please share your thoughts there. 
Closing this issue in favor of the other.
